File: Atcoder/abc315/G.cpp
#include <iostream>
#include <vector>
#include <cstring>
#include <algorithm>

using namespace std;

typedef long long ll;
const int N = 1e5 + 100;
const int MOD = 998244353;
ll n, a, b, c, x;
ll res = 0;

void exgcd(ll a, ll b, ll& x, ll& y) {
    if (b == 0ll) {
        x = 1ll;
        y = 0ll;
        return;
    }
    exgcd(b, a % b, x, y);
    ll tx = x, ty = y;
    x = ty;
    y = tx  - (a / b) * ty;
}

ll get_ans(ll a, ll b, ll c) {
    if (c <= 0) return 0;
    ll gg = __gcd(a, b);
    a /= gg;
    b /= gg;
    if (c % gg > 0) return 0;
    c /= gg;
    ll tx, ty;
    exgcd(a, b, tx, ty);
    ll maxn = min(n, (c - b) / a);
    ll minn = max(0ll, (c - n * b - 1) / a);
    ll x = c % b * tx % b;
    while (x <= 0) x += b;
    if (maxn < x) return 0;
    if (maxn <= minn) return 0;
    ll r = (maxn - x) / b + 1;
    ll l;
    if (minn < x) l = 0;
    else l = (minn - x) / b + 1;

    return r - l;
    
}

void sol() {
    cin >> n >> a >> b >> c >> x;
    for (int i = 1; i <= n; ++i) {
        res += get_ans(a, b, x - c * i);
    }
    cout << res << '\n';
}

int main() {
    // ios::sync_with_stdio(0);
    // cin.tie(0);
    // cout.tie(0);
    int T = 1;
    // cin >> T;
    while (T--) {
        sol();
    }
    exit(0);
}

File: Codeforces/div3-970-2008/G.py

def sol():
    n, k = map(int,input().split())
    a = list(map(int, input().split()))
    if n == 1:
        print(k - 1 if k - 1 < a[0] else k)
        return
    G = 0
    def gcd(a, b):
        return a if b == 0 else gcd(b, a % b)
    for i in a:
        G = gcd(G, i)
    if (n - 1) * (G - 1) < k:
        k -= (n - 1) * (G - 1)
        print((n - 1) * G + k)
    else:
        k += (k - 1) // (G - 1)
        print(k)


if __name__ == '__main__':
    t = int(input())
    while t > 0:
        t -= 1
        sol()